### PR TITLE
Source separated

### PR DESCRIPTION
**Pull request by mloubout**
Small PR to make the examples a bit more usable by people. The source and data are now separated avoiding the `None` argument at every operator call. Everything else is the same.

### CODE REVIEW

File: examples/containers.py [hunk 1]
```diff
@@ -24,12 +24,27 @@ def __init__(self, origin, spacing, vp, epsilon=None,
             self.scale = np.sqrt(1 + 2 * np.max(self.epsilon))
         else:
             self.scale = 1
+            self.epsilon = None
+
         if delta is not None:
             self.delta = np.sqrt(1 + 2 * delta)
+        else:
+            self.delta = None
```
Comment (FabioLuporini) [inline examples/containers.py]: It's already None, isn't it?

File: examples/containers.py [hunk 2]
```diff
@@ -24,12 +24,27 @@ def __init__(self, origin, spacing, vp, epsilon=None,
...
+
         if phi is not None:
             self.phi = phi
+        else:
+            self.phi = None
```
Comment (FabioLuporini) [inline examples/containers.py]: same

File: examples/containers.py [hunk 3]
```diff
@@ -24,12 +24,27 @@ def __init__(self, origin, spacing, vp, epsilon=None,
...
+
         if theta is not None:
             self.theta = theta
+        else:
+            self.theta = None
+
+        if rho is not None:
+            self.rho = rho
+        else:
+            self.rho = None
```
Comment (FabioLuporini) [inline examples/containers.py]: same

File: examples/containers.py (hunk 2 above)
Comment (FabioLuporini) [inline examples/containers.py]: isn't this basically just ``self.phi = phi``

File: tests/test_gradient.py [hunk 4]
```diff
@@ -30,16 +30,17 @@ def smooth10(vel):
         # True velocity
         true_vp = np.ones(dimensions) + .5
         if len(dimensions) == 2:
-            true_vp[:, int(dimensions[1] / 3):] = 2.5
+            true_vp[:, int(dimensions[1] / 2):] = 2
```
Comment (FabioLuporini) [inline tests/test_gradient.py]: what happened here?

File: examples/containers.py [hunk 5]
```diff
@@ -14,7 +14,7 @@ class IGrid:
     :param: theta: Tilt angle in radian
     :param phi : Asymuth angle in radian
     """
-    def __init__(self, origin, spacing, vp, epsilon=None,
+    def __init__(self, origin, spacing, vp, rho=None, epsilon=None,
```
Comment (mloubout) [inline examples/containers.py]: Good catch, done.

**Comment by FabioLuporini** [comment]
This needs rebase, and then it's probably mergeable?

**Comment by mlange05** [comment]
Yes, good to go. Merging...

**Pull request merged**